flows/mpls/mpls.go

```diff
@@ -186,6 +191,48 @@ func headers(f *otg.Flow) ([]gopacket.SerializableLayer, error) {
 	return pktLayers, nil
 }
 
+// bpfFilter generates a BPF filter that matches this flow. It returns an error if it cannot
+// build a filter.
+func bpfFilter(hdrs []gopacket.SerializableLayer) (string, error) {
+	if len(hdrs) < 2 {
+		return "", fmt.Errorf("insufficient layers to extract IPv4 headers, got %d", len(hdrs))
+	}
+	ipv4Hdr, ok := hdrs[len(hdrs)-2].(*layers.IPv4)
+	if !ok {
+		return "", fmt.Errorf("invalid headers, penultimate layer is not IPv4, got: %T", hdrs[len(hdrs)-2])
+	}
+
+	buf := &bytes.Buffer{}
+
+	// BPF when we use the 'mpls' keyword will set the offset to be +4b to look for the IP header, but this doesn't
```

Comment:
This is cool!

Reply:
Hah. It's certainly nice -- took a bit of figuring out to work out that just using `(mpls and mpls and mpls ...)` didn't do what I expected. `tcpdump -d` is a nice way to understand exactly what BPF is going to come out of these text filters.

Were there changes that you wanted here?

Reply:
No I think this is sufficient for now.

Reply:
Thanks - can you approve when you get a moment? Thanks (review is just sent with "Comment" rather than "Approval"/LGTM).